### ISSUE TITLE
Login as one user and run commands as a deploy user 

### ISSUE THREAD

**Issue by alexwebgr** [mina-deploy/mina#574]
hi guys 

i have encountered a scenario where for the sake of security commands must run as a 'deploy' user but i login as 'alex' 
how can i switch users before executing any commands and ideally without touching mina source code ?
so far i have added a new task in mina-1.0.7/tasks/mina/deploy.rb

`task :switch_user do
  command %{sudo su - deploy}
end`

which i invoke in the setup task but i when i run the mina setup it hangs. I assume this is because mina is running in a non-interactive ssh mode.

when i add this task in the generated deploy.rb setup task it runs last 

thanks 
alex

**Comment by d4be4st** [comment]
You are correct in your assumption about mina running in non-interactvie ssh mode.
And i believe it hangs because it is waiting for your to input your password.

I believe a fix would be to set execution mode to system

`set :execution_mode, :system`

Another solution would be to remove the need of a password when changing su to deploy.
Like adding this:
`username ALL=(ALL) NOPASSWD: sudo su - deploy`
to your sudoers file.

**Comment by alexwebgr** [comment]
it appears that the deploy was already set to with NOPASSWD
however after setting the execution to system now it logs in and does nothing !

**Comment by d4be4st** [comment]
We do not have this kind of setup so I can't test it unfortunately.

Will try to set something up in the coming days and see if I can find a fix/workaround for this problem

**Comment by alexwebgr** [comment]
Ok cool !

**Comment by alperenkocyigit** [comment]
I have the same issue. I even try to solve by `set :ssh_options, '-oControlPersist=no'` but it didn't work for me. When changing user cause new session and mina cant switch to that session until other one terminated.
